File: AlHadoop/src/main/java/com/al/entity/Content.java
package com.al.entity;

import java.util.HashSet;
import java.util.Set;

public class Content implements java.io.Serializable {
	private long contentId;
	private int dimeId;
	private String day;
	private int second;
	private String url;
	private String title;
	private int pv;
	private int uv;
	private Set<String> uuids;
	
	public void addReuqest(String uuid) {
		pv++;
		if(uuids == null) {
			uuids = new HashSet<String>();
		}
		uuids.add(uuid);
		uv = uuids.size();
	}

	public long getContentId() {
		return contentId;
	}
	public void setContentId(long contentId) {
		this.contentId = contentId;
	}
	public int getDimeId() {
		return dimeId;
	}
	public void setDimeId(int dimeId) {
		this.dimeId = dimeId;
	}
	public String getDay() {
		return day;
	}
	public void setDay(String day) {
		this.day = day;
	}
	public int getSecond() {
		return second;
	}
	public void setSecond(int second) {
		this.second = second;
	}
	public String getUrl() {
		return url;
	}
	public void setUrl(String url) {
		this.url = url;
	}
	public String getTitle() {
		return title;
	}
	public void setTitle(String title) {
		this.title = title;
	}
	public int getPv() {
		return pv;
	}
	public void setPv(int pv) {
		this.pv = pv;
	}
	public int getUv() {
		return uv;
	}
	public void setUv(int uv) {
		this.uv = uv;
	}
	public Set<String> getUuids() {
		return uuids;
	}
	public void setUuids(Set<String> uuids) {
		this.uuids = uuids;
	}
	
}

File: AlSpark/src/main/scala/com/al/entity/Content.scala
package com.al.entity

import scala.beans.BeanProperty

/**
  * Created by An on 2016/11/25.
  */
class Content extends Serializable {
  @BeanProperty var contentId: Long = 0
  @BeanProperty var url: String = ""
  @BeanProperty var title: String = ""
  @BeanProperty var day: String = ""
  @BeanProperty var pv: Int = 0
  @BeanProperty var uv: Int = 0

  @BeanProperty var dimeId: Int = 0
  @BeanProperty var second: Int = 0

  var uvs = Set[String]()

  def this(pv: Int, uv: Int){
    this()
    this.pv = pv
    this.uv = uv
  }
}
